internal/driver/driver.go

```diff
@@ -226,32 +226,57 @@ func (d *Driver) HandleReadCommands(deviceName string, protocols map[string]mode
 	reqs []sdkModels.CommandRequest) ([]*sdkModels.CommandValue, error) {
 	d.lc.Debugf("Driver.HandleReadCommands: protocols: %v resource: %v attributes: %v", protocols,
 		reqs[0].DeviceResourceName, reqs[0].Attributes)
-	var err error
 	var responses = make([]*sdkModels.CommandValue, len(reqs))
 
 	device, edgexErr := d.getDevice(deviceName)
 	if edgexErr != nil {
 		return responses, edgexErr
 	}
-	// currently defaults to using the first available stream
-	cameraDevice, err := usbdevice.Open(device.paths[0])
-	if err != nil {
-		return responses, errors.NewCommonEdgeX(errors.KindServerError,
-			fmt.Sprintf("failed to open the underlying device at specified path %s", device.paths[0]), err)
-	}
-	defer cameraDevice.Close()
 
 	var cv *sdkModels.CommandValue
 	var data interface{}
 	errorWrapper := EdgeXErrorWrapper{}
 	for i, req := range reqs {
-		command, ok := req.Attributes[Command]
+		command, ok := req.Attributes[GetFunction]
 		if !ok {
 			return responses, errors.NewCommonEdgeX(errors.KindContractInvalid,
 				fmt.Sprintf("command for USB camera resource %s is not specified, please check device profile",
 					req.DeviceResourceName), nil)
 		}
 
+		queryParams, edgexErr := getQueryParameters(req)
+		if edgexErr != nil {
+			return responses, errors.NewCommonEdgeXWrapper(edgexErr)
+		}
+
+		if _, ok := req.Attributes[UrlRawQuery]; ok {
+			req.Attributes[UrlRawQuery] = ""
+		}
+
+		var videoPath string
+		pathIndex := queryParams.Get(PathIndex)
+		if len(pathIndex) == 0 {
+			videoPath = device.paths[0]
+		} else {
+			pathIndexConv, err := strconv.Atoi(pathIndex)
+			if err != nil {
+				return nil, err
+			}
+			if pathIndexConv >= len(device.paths) {
+				return nil, errors.NewCommonEdgeX(errors.KindIOError,
+					fmt.Sprintf("Video streaming path does not exist for the device %v at PathIndex %d", device.name, pathIndexConv), nil)
+			}
+			videoPath = device.paths[pathIndexConv]
+		}
+
+		// currently defaults to using the first available stream
+		cameraDevice, err := usbDevice.Open(videoPath)
+		if err != nil {
+			return responses, errors.NewCommonEdgeX(errors.KindServerError,
+				fmt.Sprintf("failed to open the underlying device at specified path %s", videoPath), err)
+		}
+		defer cameraDevice.Close()
```

Comment:
I see warning in my IDE for this for a possible resource leak.

Reply:
Does it give more details? This is the standard practice for opening and closing the device across the service

Reply:
`Possible resource leak, 'defer' is called in the 'for' loop`

Reply:
I think defer should be removed and it should be just `cameraDevice.Close()` or you probably need to move this part of opening and closing into a separate function.

Reply:
yes, what will happen is the defer is not called until the entire for loop is completed, where what you really want is it to be closed at the end of each for iteration.

this can be addressed by either not using defer and closing it at the end (if applicable), or moving part of the body of the for loop into a separate function that can have a defer inside of it.